vignettes/datatable-programming.Rmd

```diff
@@ -373,6 +373,29 @@ print(j)
 DT[, j, env = list(j = j)]
 ```
 
+### Injecting functions using strings in `env`
+
+In `data.table`, you can inject functions into your expressions by passing their names as strings in the `env` parameter. This method allows you to use function names directly as strings, and `data.table` will automatically interpret these strings as the corresponding functions. This approach simplifies the process and makes it easier to work with function names dynamically.
+
+Suppose you want to calculate the total of `Sepal.Length` in the `iris` dataset using the `sum` function. You can inject the `sum` function by passing its name as a string in the `env` parameter.
+
+```{r sum_example}
+result <- DT[, f(Sepal.Length), env = list(f = "sum")]
+print(result)
+```
+
+In this example, the string `"sum"` is mapped to the `f` function. The `env` parameter allows `data.table` to recognize this `f` as the `sum` function, so it computes the total of `Sepal.Length` accordingly.
```

Comment:
this is ok for a first example, but I believe the second example below could be better, by showing a more typical use case, such as
```r
my_summarization <- function(DT, f, variable, by){
  DT[, f(variable), by=by, env=list(f=f, variable=variable, by=by)]
}
```

Reply:
Toby's example without it's usage is not enough. Try with `by` of length 2+